src/cli_util.rs

```diff
@@ -1234,6 +1234,46 @@ pub fn write_commit_summary(
     Ok(())
 }
 
+pub fn write_config_entry(
+    ui: &mut Ui,
+    path: &str,
+    value: config::Value,
+) -> Result<(), CommandError> {
+    match value.kind {
+        config::ValueKind::Table(table) => {
+            for (key, table_val) in table {
+                let key_path = match path {
+                    "" => key,
+                    _ => format!("{path}.{key}"),
+                };
+                write_config_entry(ui, key_path.as_str(), table_val)?;
+            }
+        }
+        _ => writeln!(ui, "{path}={}", serialize_config_value(value))?,
+    };
+    Ok(())
+}
+
+fn serialize_config_value(value: config::Value) -> String {
+    match value.kind {
+        config::ValueKind::Table(table) => format!(
+            "{{{}}}",
+            table
+                .into_iter()
+                .map(|(k, v)| format!("{k}={}", serialize_config_value(v)))
+                .join(", ")
+        ),
+        config::ValueKind::Array(vals) => format!(
+            "[{}]",
+            vals.into_iter()
+                .map(|v| serialize_config_value(v))
+                .join(", ")
+        ),
+        config::ValueKind::String(val) => format!("{val:#?}"),
```

Comment:
Is the `:#?` format different from the `:?` format for strings?

Reply:
You're right, `:?` is better.